### PR TITLE
Update wgpu-rs to reflect latest wgpu API

### PR DESCRIPTION
**Pull request by kyren**
Not sure if everything here is correct, especially `RenderPassColorAttachmentDescriptor` and the `resolve_target` field.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -161,7 +160,7 @@ pub struct Queue<'a> {
 pub enum BindingResource<'a> {
     Buffer {
         buffer: &'a Buffer,
-        range: Range<u32>,
+        range: Range<u64>,
```
Comment (kvark) [inline src/lib.rs]: we should use the [BufferAddress](https://github.com/gfx-rs/wgpu/blob/a056d2478a3e34fa20071957af89fc6eda7a313d/wgpu-native/src/lib.rs#L53) alias for these

File: src/lib.rs [hunk 2]
```diff
@@ -726,8 +733,9 @@ impl CommandEncoder {
         let colors = desc
             .color_attachments
             .iter()
-            .map(|ca| RenderPassColorAttachmentDescriptor {
+            .map(|ca| wgn::RenderPassColorAttachmentDescriptor {
                 attachment: ca.attachment.id,
+                resolve_target: ca.resolve_target.map(|v| &v.id as *const _).unwrap_or(ptr::null()),
```
Comment (kvark) [inline src/lib.rs]: nit: map_or()

File: src/lib.rs [hunk 3]
```diff
@@ -216,11 +215,19 @@ pub struct ComputePipelineDescriptor<'a> {
...
 pub struct RenderPassDescriptor<'a> {
-    pub color_attachments: &'a [RenderPassColorAttachmentDescriptor<&'a TextureView>],
+    pub color_attachments: &'a [RenderPassColorAttachmentDescriptor<'a, &'a TextureView>],
     pub depth_stencil_attachment:
         Option<RenderPassDepthStencilAttachmentDescriptor<&'a TextureView>>,
 }
 
+pub struct RenderPassColorAttachmentDescriptor<'a, T> {
+    pub attachment: T,
+    pub resolve_target: Option<&'a TextureView>,
```
Comment (kvark) [inline src/lib.rs]: let's use `Option<T>` here

File: src/lib.rs [hunk 4]
```diff
@@ -441,7 +448,7 @@ impl Device {
                         module: desc.vertex_stage.module.id,
                         entry_point: vertex_entry_point.as_ptr(),
                     },
-                    fragment_stage: wgn::PipelineStageDescriptor {
+                    fragment_stage: &wgn::PipelineStageDescriptor {
                         module: desc.fragment_stage.module.id,
```
Comment (kvark) [inline src/lib.rs]: we should make `desc.fragment_stage` to also be an `Option` now

File: src/lib.rs (hunk 2 above)
Comment (kyren) [reply src/lib.rs]: yep, derp.

File: src/lib.rs (hunk 1 above)
Comment (kyren) [reply src/lib.rs]: Are there other instances besides this one that should use the BufferAddress alias?

File: src/lib.rs (hunk 4 above)
Comment (kyren) [reply src/lib.rs]: Yeah that should have been obvious, sorry!

File: src/lib.rs [hunk 5]
```diff
@@ -418,7 +426,20 @@ impl Device {
...
+        let (_fragment_entry_point, fragment_stage) = if let Some(fragment_stage) = &desc.fragment_stage {
+            let fragment_entry_point = CString::new(fragment_stage.entry_point).unwrap();
+            let fragment_stage = wgn::PipelineStageDescriptor {
+                module: fragment_stage.module.id,
+                entry_point: fragment_entry_point.as_ptr(),
+            };
+            (fragment_entry_point, Some(fragment_stage))
+        } else {
+            (CString::default(), None)
+        };
```
Comment (kyren) [inline src/lib.rs]: I'm uncomfortable with how not-obviously-safe this is, but I couldn't think of a better way :/

File: src/lib.rs (hunk 1 above)
Comment (kyren) [reply src/lib.rs]: Yeah okay, it's a lot more, it would have been obvious had I actually looked at wgpu more closely!

**Comment by kyren** [comment]
> sorry about the inconvenience

Not inconvenient at all!

File: src/lib.rs (hunk 1 above)
Comment (kyren) [reply src/lib.rs]: Actually just %s/u64/BufferAddress/g was correct, I made it harder than it had to be :P

**Pull request merged**